### ISSUE TITLE
Great work Finetune Scripts

### ISSUE THREAD

**Issue by Oddadmix** [SWivid/Habibi-TTS#2]
Great work on the model, it will be a great base model.

Would you be able to share the finetuning script to furthur improve it for a particular dialact.

Thanks in advance.

**Comment by SWivid** [comment]
Hi @Oddadmix , the whole training/finetuning framework is with https://github.com/SWivid/F5-TTS?tab=readme-ov-file#training

Specifically, our own approach is using train.py, step-by-step:
First of all, clone the F5-TTS repo and install editable, and make sure you are currently at the main directory (`cd F5-TTS`)
(**you could use finetune_gradio.py for sure if wish, but remember to follow the same hyperparam setup**)

1. Prepare your train set (will be a folder under F5-TTS/data/), some examples scripts: https://github.com/SWivid/F5-TTS/tree/main/src/f5_tts/train/datasets; we use "char" tokenizer here for Arabic. 
Note that if you also want to use the regional identifier as mentioned in [our paper](https://arxiv.org/abs/2601.13802), during the preprocess, need to wrap the text with corresponding id, see https://github.com/SWivid/Habibi-TTS/blob/64170e88a4f33cf8bb3623c88ba3cba033fdbb29/src/habibi_tts/model/utils.py#L1-L26 And as you can see in [the vocab.txt of pretrained unified model](https://huggingface.co/SWivid/Habibi-TTS/blob/main/Unified/vocab.txt#L2721-L2728), we have reserved some id (13 to 20 is not used yet).
If you want to finetune based on the pretrained model, remember to replace the vocab.txt with the one of pretrained (https://huggingface.co/SWivid/Habibi-TTS/blob/main/Unified/vocab.txt). 
Furthermore, if need to extend the vocab and text_embed correspondingly, draft your own script to do so (quite easy, and generally speaking, not necessary; [some code for reference](https://github.com/SWivid/F5-TTS/blob/655fbca5529b68e36d7b8f2d6d131f68132156fe/src/f5_tts/train/finetune_gradio.py#L957-L1108)).

2. Make your config yaml under https://github.com/SWivid/F5-TTS/tree/main/src/f5_tts/configs, could copy the v1_base one as template to start with:
   2.1. Change the [datasets-name](https://github.com/SWivid/F5-TTS/blob/655fbca5529b68e36d7b8f2d6d131f68132156fe/src/f5_tts/configs/F5TTS_v1_Base.yaml#L6) (not to include the tokenizer postfix, just the name);
   2.2. Set epochs (could use [this script](https://github.com/SWivid/F5-TTS/blob/main/src/f5_tts/scripts/count_max_epoch_precise.py) to calculate, we generally set the max_updates=200000);
   2.3. Set your [experiment/model name](https://github.com/SWivid/F5-TTS/blob/655fbca5529b68e36d7b8f2d6d131f68132156fe/src/f5_tts/configs/F5TTS_v1_Base.yaml#L21), and [tokenizer](https://github.com/SWivid/F5-TTS/blob/655fbca5529b68e36d7b8f2d6d131f68132156fe/src/f5_tts/configs/F5TTS_v1_Base.yaml#L22) to "char";
   2.4. Others we keep the default (e.g. learning rate=7.5e-5, ...)

3. If you want to finetune based on the pretrained model, just 
   3.1. Download the checkpoint (.safetensors, e.g. [the unified one](https://huggingface.co/SWivid/Habibi-TTS/blob/main/Unified/model_200000.safetensors)) from huggingface;
   3.2. Rename as pretrained_xxx (e.g. pretrained_unified_model_200000.safetensors) and create the folder and place under [here](https://github.com/SWivid/F5-TTS/blob/655fbca5529b68e36d7b8f2d6d131f68132156fe/src/f5_tts/configs/F5TTS_v1_Base.yaml#L55) (`ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}`, where `${}` corresponds to the value you filled in the yaml file)

4. Ready to go now, (see [more](https://github.com/SWivid/F5-TTS/tree/main/src/f5_tts/train#1-training-script-used-for-pretrained-model))
    ```bash
    # setup accelerate config, we use multi-gpu ddp, fp16, others default
    accelerate config
    
    # .yaml files are under src/f5_tts/configs directory
    accelerate launch src/f5_tts/train/train.py --config-name YOUR_YAML_FILE_NAME.yaml
    # training/finetuning will start, checkpoints will be under ckpts/xxx
    ```
